graylog2-web-interface/src/components/grok-patterns/GrokPatterns.jsx

```diff
@@ -22,60 +24,100 @@ const GrokPatternsList = styled(DataTable)`
   }
 `;
 
-const GrokPatterns = createReactClass({
-  displayName: 'GrokPatterns',
+class GrokPatterns extends React.Component {
+  constructor(props) {
+    super(props);
 
-  getInitialState() {
-    return {
+    this.state = {
       patterns: [],
+      pagination: {
+        page: 1,
+        perPage: 10,
+        count: 0,
+        total: 0,
+        query: '',
+      },
     };
-  },
+  }
 
   componentDidMount() {
     this.loadData();
-  },
+  }
 
   componentWillUnmount() {
     if (this.loadPromise) {
       this.loadPromise.cancel();
     }
-  },
+  }
 
-  loadData() {
-    this.loadPromise = GrokPatternsStore.loadPatterns((patterns) => {
-      if (!this.loadPromise.isCancelled()) {
-        this.loadPromise = undefined;
+  loadData = (callback) => {
+    const { pagination: pagination1 } = this.state;
+    const { page, perPage, query } = pagination1;
 
-        this.setState({
-          patterns: patterns,
-        });
-      }
-    });
-  },
+    this.loadPromise = GrokPatternsStore.searchPaginated(page, perPage, query)
+      .then(({ patterns, pagination }) => {
+        if (callback) {
+          callback();
+        }
+
+        if (!this.loadPromise.isCancelled()) {
+          this.loadPromise = undefined;
 
-  validPatternName(name) {
+          this.setState({
+            patterns: patterns,
```

Comment:
Match code style with line 68
```suggestion
            patterns,
```